apps/native-component-list/src/screens/Video/VideoScreen.tsx

```diff
@@ -102,17 +116,30 @@ export default function VideoScreen() {
   );
 
   const updatePreservesPitch = useCallback(
-    (correctPitch: boolean) => {
-      player.preservesPitch = correctPitch;
-      setCorrectsPitch(correctPitch);
+    (preservesPitch: boolean) => {
+      player.preservesPitch = preservesPitch;
+      setPreservePitch(preservesPitch);
     },
     [player]
   );
 
   useEffect(() => {
-    player.play();
-    player.preservesPitch = shouldCorrectPitch;
-  }, [player]);
+    if (logEvents) {
+      eventsToListen.map((eventName) => {
+        player.addListener(eventName, (newValue: any, _: any, error: any) => {
```

Comment:
`addListener` returns a subscription object that has a `remove` function. I think it might be more correct to map `eventsToListen` to an array of subscriptions and then use that function instead of removing all listeners (although removing them all is for sure more efficient!).

Also, if you're not mapping and just need to iterate through, it might be better to use `forEach` or a tranditional `for of` 😉

Reply:
I'll stick to `removeAllListeners` for now, we need to [update ts-declarations for EventEmitter](https://github.com/expo/expo/blob/0441c90af80bea6036f3eadcef043d63020ed9b8/packages/expo-modules-core/src/ts-declarations/EventEmitter.ts#L34C2-L37C11), but I don't want to do it in this PR, as it already contains a ton of changes